packages/core/src/node/backend-application.ts

```diff
@@ -150,6 +151,13 @@ export class BackendApplication {
             server = http.createServer(this.app);
         }
 
+        server.on('error', error => {
+            deferred.reject(error);
+            /* The backend might run in a separate process,
+             * so we defer `process.exit` to let time for logging in the parent process */
+            setTimeout(process.exit, 0, 1);
```

Comment:
What logging do you mean?

Reply:
Any logging, errors or messages thrown by the process aren't channelled through the root terminal process if we don't let just a bit of time pass. I don't clearly understand why it happens, but my guess would be that the IPC channels that bubble errors up might be registered in the eventloop, and registering `process.exit` at the end of the event stack might allow the logging to be sent ?

Reply:
_(if I don't do this `setTimeout(exit, 0ms, ...)` no messages get logged in the root terminal used to start Theia; `setImmediate(...)` doesn't work either)_

Reply:
_(if you want an example, try to run 2 Theias instance on the same endpoint, this will throw an error)_

Reply:
Maybe we should enrich the protocol that the server worker asks the master process to stop it then the master will have all control.